libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2245,6 +2245,22 @@ void TypeChecker::typeCheckFunctionGeneralChecks(
 		}
 	}
 
+	if (_functionType->kind() == FunctionType::Kind::BytesConcat)
+	{
+		solAssert(paramArgMap.empty(), "");
+		for (auto const& argument: arguments)
+			if (
+				Type const* argumentType = type(*argument);
```

Comment:
Will this change to using `isImplicitlyConvertibleTo` later?

Reply:
I was thinking to add it immediately, but wasn't sure what should I use for target type for conversion in that case? `bytes32`?

Reply:
Yes, try `bytes32` and `bytesMemory`.

Reply:
Fixed